src/View/Antlers/Language/Analyzers/RecursiveParentAnalyzer.php

```diff
@@ -53,6 +53,12 @@ public static function associateRecursiveParent($nodes)
                         } else {
                             if (Str::contains($subNode->runtimeContent, $recursiveContent) && mb_substr_count($subNode->runtimeContent, '*recursive') == 1 && $node->getRootRef() == $subNode->getRootRef()) {
                                 $lastNode = $subNode;
+
+                                // Ensure we stop searching once we reach the closest nav parent.
+                                if ($subNode->name->name == 'nav') {
```

Comment:
Not that we have any other tag with recursion... but won't this only fix the `nav` tag specifically?

(I'm fine with this, though)

Reply:
Yea, this will be specific to the `nav` tag, unfortunately. I will do some more thinking on how to fix it a general way, but that will be a bit more challenging/difficult.

Reply:
That's okay, this is fine.